src/field_theory/finite.lean

```diff
@@ -9,15 +9,40 @@ import data.zmod.basic
 import linear_algebra.basis
 import ring_theory.integral_domain
 
-universes u v
-variables {K : Type u} {R : Type v}
+/-!
+# Finite fields
 
-open function finset polynomial nat
+This file contains basic results about finite fields.
+Throughout most of this file, `K` denotes a finite field with `q` elements.
+
+## Main results
+
+1. Every finite integral domain is a field (`field_of_integral_domain`).
+2. The unit group of a finite field is a cyclic group of order `q - 1`.
+   (`finite_field.is_cyclic` and `card_units`)
+3. `sum_pow_units`: The sum of `x^i`, where `x` ranges over the units of `K`, is
+   - `q-1` if `q-1 ∣ i`
+   - `0`   otherwise
+4. `finite_field.card`: The cardinality `q` is a power of the characteristic of `K`.
+   See `card'` for a variant.
+-/
+
+variables {K : Type*} [field K] [fintype K]
+variables {R : Type*} [integral_domain R]
+local notation `q` := fintype.card K
+
+open_locale big_operators
 
 namespace finite_field
+open function finset polynomial
+
+/-- The unit group of a finite integral domain is cyclic. -/
+instance [fintype R] : is_cyclic (units R) :=
+let φ : units R →* R := { to_fun := coe, map_one' := units.coe_one, map_mul' := units.coe_mul } in
```

Comment:
This guy probably belongs somewhere else (`algebra.group.units`?).

Reply:
Moved to the file on integral domains.